apkid/rules/elf/obfuscators.yara

```diff
@@ -155,6 +155,53 @@ rule ollvm : obfuscator
     not ollvm_v6_0_strenc
 }
 
+rule byteguard_0_9_3 : obfuscator
+{
+  meta:
+    description = "ByteGuard 0.9.3"
+    sample      = "eed4f7b907fe2173935d307dfb0d6aa7098f69db8dfb65e49affd7b7a6c0a5e4"
+    samples     = "https://koodous.com/rulesets/5862/apks"
+    author      = "Eduardo Novella"
+
+  strings:
+    // clang version 6.0.0 (Byteguard 0.6) (git@sysrepo.byted.org:dingbaozeng/native_obfuscator.git 448f20ff6eb06dd336dd81846d6a7dc8ba8c961b)
+    // Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)
+    $version = "Apple LLVM version 6.0.0 (ByteGuard 0.9.3-af515063)"
+
+  condition:
+    is_elf and all of them
+}
+
+rule byteguard_0_9_2 : obfuscator
+{
+  meta:
+    description = "ByteGuard 0.9.2"
+    sample      = "178b1ef3c4ac563604c8a262f0e3651f56995768c8aa13ccc845f33bd6eb0ac2"
+    samples     = "https://koodous.com/rulesets/5862/apks"
+    author      = "Eduardo Novella"
+
+  strings:
+    $clang_version = "clang version 5.0.2 (Byteguard-0.9.2-255c7b5e)"
+
+  condition:
+    is_elf and all of them
+}
+
+rule byteguard_unknown : obfuscator
+{
+  meta:
+    description = "ByteGuard unknown version"
+    author      = "Eduardo Novella"
+
+  strings:
+    $clang_version = /clang version \d\.\d\.\d \(Byteguard(.*)0\.\d\.\d\-(.*)\)/
```

Comment:
Thought that strings were faster than regex... All suggestions were applied.

Reply:
```sh
[*] 178b1ef3c4ac563604c8a262f0e3651f56995768c8aa13ccc845f33bd6eb0ac2.apk!lib/armeabi-v7a/libcjtfcc.so
 |-> obfuscator : ByteGuard 0.9.2
[*] 178b1ef3c4ac563604c8a262f0e3651f56995768c8aa13ccc845f33bd6eb0ac2.apk!lib/armeabi-v7a/libcms.so
 |-> obfuscator : ByteGuard 0.9.3
```

Reply:
Should we merge 0.9.2 and 0.9.3 into 0.9.x? The same APK seems obfuscated with slightly different versions.

Reply:
Strings are faster than regex but I think it's worth it for robustness.

No need to merge. I think it's good to fingerprint patch versions if we already have the rules to do it. I was suggesting a 0.9.x version on top of the others, but it's not that important since it should be caught by the unknown rule.